mayastor-test/e2e/uninstall/uninstall_test.go

```diff
@@ -0,0 +1,193 @@
+/// the IP address of the master node (if one exists),
+/// The assumption is that the test-registry is accessible via the IP addr of the master,
+/// or any node in the cluster if the master noe does not exist
+/// TODO Refine how we workout the address of the test-registry
+func getRegistryAddress() (string, error) {
+	var master = ""
+	nodeList := coreV1.NodeList{}
+	if (k8sClient.List(context.TODO(), &nodeList, &client.ListOptions{}) != nil) {
+		return master, errors.New("failed to list nodes")
+	}
+	nodeIPs := make([]string, len(nodeList.Items))
+	for ix, k8node := range nodeList.Items {
+		for _, k8Addr := range k8node.Status.Addresses {
+			if k8Addr.Type == coreV1.NodeInternalIP {
+				nodeIPs[ix] = k8Addr.Address
+				for label := range k8node.Labels {
+					if label == "node-role.kubernetes.io/master" {
+						master = k8Addr.Address
+					}
+				}
+			}
+		}
+	}
+
+	/// TODO Refine how we workout the address of the test-registry
+
+	/// If there is master node, use its IP address as the registry IP address
+	if len(master) != 0 {
+		return master, nil
+	}
+
+	if len(nodeIPs) == 0 {
+		return "", errors.New("no usable nodes found")
+	}
+
+	/// Choose the IP address of first node in the list as the registry IP address
+	return nodeIPs[0], nil
+}
+
+// Encapsulate the logic to find where the deploy yamls are
+func getDeployYamlDir() string {
+	_, filename, _, _ := runtime.Caller(0)
+	return path.Clean(filename + "/../../../../deploy")
+}
+
+// Helper for passing yaml from the deploy directory to kubectl
+func deleteDeployYaml(filename string) {
+	cmd := exec.Command("kubectl", "delete", "-f", filename)
+	cmd.Dir = getDeployYamlDir()
+	_, err := cmd.CombinedOutput()
+	Expect(err).ToNot(HaveOccurred())
+}
+
+// Encapsulate the logic to find where the templated yamls are
+func getTemplateYamlDir() string {
+	_, filename, _, _ := runtime.Caller(0)
+	return path.Clean(filename + "/../../install/deploy")
+}
+
+func makeImageName(registryAddress string, registryport string, imagename string, imageversion string) string {
+	return registryAddress + ":" + registryport + "/mayadata/" + imagename + ":" + imageversion
+}
+
+func deleteTemplatedYaml(filename string, imagename string, registryAddress string) {
+	fullimagename := makeImageName(registryAddress, "30291", imagename, "ci")
+	bashcmd := "IMAGE_NAME=" + fullimagename + " envsubst < " + filename + " | kubectl delete -f -"
+	cmd := exec.Command("bash", "-c", bashcmd)
+	cmd.Dir = getTemplateYamlDir()
+	_, err := cmd.CombinedOutput()
+	Expect(err).ToNot(HaveOccurred())
+}
+
+// We expect this to fail a few times before it succeeds,
+// so no throwing errors from here.
+func mayastorReadyPodCount() int {
+	var mayastorDaemonSet appsv1.DaemonSet
+	if k8sClient.Get(context.TODO(), types.NamespacedName{Name: "mayastor", Namespace: "mayastor"}, &mayastorDaemonSet) != nil {
+		return -1
+	}
+	return int(mayastorDaemonSet.Status.CurrentNumberScheduled)
+}
```

Comment:
As Chris already mentioned, I think functions like this should eventually go in a common file, as we are likely to need them in other tests.

Reply:
Sure, I will do that as another PR, there is a bunch of stuff that needs to be moved from the install test as well.